src/Features/CSharp/Portable/CodeFixes/HideBase/HideBaseCodeFixProvider.AddNewKeywordAction.cs

```diff
@@ -25,16 +33,36 @@ protected override async Task<Document> GetChangedDocumentAsync(CancellationToke
             {
                 var root = await _document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
-                var newNode = GetNewNode(_node);
+                var newNode = await GetNewNode(_node, cancellationToken).ConfigureAwait(false);
                 var newRoot = root.ReplaceNode(_node, newNode);
 
                 return _document.WithSyntaxRoot(newRoot);
             }
 
-            private SyntaxNode GetNewNode(SyntaxNode node)
+            private async Task<SyntaxNode> GetNewNode(SyntaxNode node, CancellationToken cancellationToken)
             {
                 var generator = SyntaxGenerator.GetGenerator(_document);
-                return generator.WithModifiers(node, generator.GetModifiers(node).WithIsNew(true));
+                var newNode = generator.WithModifiers(node, generator.GetModifiers(node).WithIsNew(true));
+
+                var options = await _document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+                var option = options.GetOption(PreferredModifierOrderOption);
+                if (!OrderModifiersHelper.TryGetOrComputePreferredOrder(option.Value, out var preferredOrder))
+                {
+                    return newNode;
+                }
+
+                var modifiers = SyntaxFacts.GetModifiers(newNode);
+                var orderedModifiers = new SyntaxTokenList(
+                    modifiers.OrderBy(CompareModifiers)
+                             .Select((t, i) => t.WithTriviaFrom(modifiers[i])));
```

Comment:
💭 For the specific case where we only order the modifiers if they were already ordered at the start of this method, it seems like this `Select` statement is not necessary. Since we are inserting the `new` modifier and keeping all other modifiers the same, we should be able to keep all trivia associated with the original nodes.